.github/workflows/upgrade.yml

```diff
@@ -11,7 +11,7 @@ jobs:
   build-and-push-ecr:
```

Comment:
feel like we should just call out to a script given how complex this is becoming. food for though, so feel free to disregard if you disagree

Reply:
The only portion of this that isn't handled by the actions themselves is constructing the matrix and checking for rollbacks. I'm comfortable with this complexity being handled by us, and deferring as much as possible to widely-used actions that we don't have to maintain ourselves. Plus, getting parallelism out of the matrix strategy is neat. I'll keep it this way for now, and revisit if/when we try to optimize CI